### DISCUSSION TITLE
how to show number on image-slider-response scale?

### DISCUSSION THREAD

**Discussion by ricardojromeu** [jspsych/jsPsych#1267] [Q&A] [answered]
Hello,

I would like the participant to see the number that corresponds with the position on the slider, but I'm not sure how to make this happen. I tried implemented the solution that was posted for html-slider-response, but that didn't work for me. The console gives me an error that I have an unexpected identifier in my string. Following the example in the html-slider-response solution, I tried doing this:

`prompt: "<p>Confidence for BLUE JAR: <span id="answer-display"> </span> </p>",`

And then when I push the trial to the main timeline, I did it like this:

```js
Beads_timeline.push(beads1,
    {
  type: 'image-slider-response',
  html: '', // your stimulus
  on_load: function()  {
    var input = document.querySelector('#jspsych-image-slider-response-response')

    input.addEventListener('input', function() {
      document.querySelector('#answer-display').innerText = input.valueAsNumber.toString()
    })

  }
    }
)
```

but I get the unexpected identifier error on the line with the prompt. 

Any help to make this work out would be greatly appreciated.

thanks,
-Ricky

**Comment by gorkang** [comment, marked as answer]
Hi @ricardojromeu 

Please find attached a modified version of the image plugin where you can use the parameter` slider_number: true` to show a number to the right of the slider. You can run the example in the attachment where the 1st image shows the slide.

[jspsych-image-slider.zip](https://github.com/jspsych/jsPsych/files/5649273/jspsych-image-slider.zip)

@becky-gilbert , do you think it is worth it to add this functionality to the different slider plugins and include that in the pull request #1212 ?

**Reply by ricardojromeu** [reply]
Thank you, it works great!

**Reply by becky-gilbert** [reply]
@gorkang just to clarify - do you mean adding the `slider_number` parameter to the image-slider-response plugin in your pull request for html-slider-response (#1212)? If so then yes, absolutely. It'd be great if your pull request included this option in all of the slider plugins: html-slider-response, image-slider-response, audio-slider-response, video-slider-response. But I don't know how much work that would be, so no worries if you don't have time. At the moment we're planning on adding your `slider_number` parameter to the 7.1 release, which should come out sometime in the spring/summer.

**Reply by gorkang** [reply]
Hi @becky-gilbert . Yes, given the internals of the slider plugins should be similar (I guess), I can try extending the `slider_number` parameter to all of them (and update the tests). I'll give it a try in the next few weeks and update the Pull request if I am successful. Thanks!
